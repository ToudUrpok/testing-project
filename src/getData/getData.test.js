const getData = require('./getData');
const axios = require('axios');

jest.mock('axios');

describe('getData module testing', () => {
    let response;

    beforeEach(() => {
        response = {
            data: [
                {
                    "userId": 1,
                    "id": 1,
                    "title": "delectus aut autem",
                    "completed": false
                },
                {
                    "userId": 1,
                    "id": 2,
                    "title": "quis ut nam facilis et officia qui",
                    "completed": false
                },
                {
                    "userId": 1,
                    "id": 3,
                    "title": "fugiat veniam minus",
                    "completed": false
                },
                {
                    "userId": 1,
                    "id": 4,
                    "title": "et porro tempora",
                    "completed": true
                },
                {
                    "userId": 1,
                    "id": 5,
                    "title": "laboriosam mollitia et enim quasi adipisci quia provident illum",
                    "completed": false
                }
            ]
        };
    });

    test('mock return value', async () => {
        axios.get.mockReturnValue(response);
        const data = await getData();
        expect(axios.get).toBeCalledTimes(1);
        expect(data).toMatchSnapshot();
    });
});