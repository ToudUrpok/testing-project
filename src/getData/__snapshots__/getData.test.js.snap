// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData module testing async function test 1`] = `
[
  "1",
  "2",
  "3",
  "4",
  "5",
]
`;

exports[`getData module testing mock return value 1`] = `
[
  "1",
  "2",
  "3",
  "4",
  "5",
]
`;
